### PR TITLE
[MM-34798] Respect MM_SERVER_PATH when looking for templates

### PR DESCRIPTION
**Pull request by hanzei**
#### Summary
`MM_SERVER_PATH` is an environment variables that can be set to the path of the `mattermost-server` directory and enabled running the server from a different directory then the root of the repository. When looking for the `templates´ directory the `MM_SERVER_PATH` was not taking into account. This PR fixes that.

I stumbled across this because the [Apps plugin](https://github.com/mattermost/mattermost-plugin-apps) uses `api4.Setup` to run e2e tests.

#### Ticket Link
https://mattermost.atlassian.net/browse/MM-34798

#### Release Note
```release-note
NONE
```

### CODE REVIEW

**Review by streamer45 — APPROVED** [review]
Nice :+1:

**Review by jespino — COMMENTED** [review]
Nice!

**Review by jespino — APPROVED** [review]

**Comment by hanzei** [comment]
@jespino @streamer45 Who would be a good person to QA review this?

**Comment by streamer45** [comment]
> @jespino @streamer45 Who would be a good person to QA review this?

I'd go with Saturn.

**Comment by saturninoabril** [comment]
Oopss, sorry for the delay. Will look into this today.

File: shared/templates/templates.go [hunk 1]
```diff
@@ -29,6 +31,15 @@ type Data struct {
 	HTML  map[string]template.HTML
 }
 
+func GetTemplateDirectory() (string, bool) {
+	translationsDir := "templates"
```
Comment (saturninoabril) [inline shared/templates/templates.go:35]: Sounds confusing. Maybe `templatesDir`?

**Review by saturninoabril — APPROVED** [review]
Thanks @hanzei! Tested and passed. LGTM.

File: shared/templates/templates.go (hunk 1 above)
Comment (hanzei) [reply shared/templates/templates.go:35]: Good suggestion :+1:

**Pull request merged**